xskillscore/core/probabilistic.py

```diff
@@ -352,3 +360,148 @@ def threshold_brier_score(
+):
+    """Returns the data required to construct the discrimination diagram for an event; the \
+            histogram of forecasts likelihood when observations indicate an event has occurred \
+            and has not occurred.
+
+        Parameters
+        ----------
+        observations : xarray.Dataset or xarray.DataArray
+            The observations or set of observations of the event. Data should be boolean or logical \
+            (True or 1 for event occurance, False or 0 for non-occurance).
+        forecasts : xarray.Dataset or xarray.DataArray
+            The forecast likelihoods of the event. Data should be between 0 and 1.
+        dim : str or list of str, optional
+            Dimension(s) over which to compute the histograms
+            Defaults to None meaning compute over all dimensions.
+        probability_bin_edges : array_like, optional
+            Probability bin edges used to compute the histograms. Bins include the left most edge, \
+            but not the right. Defaults to 6 equally spaced edges between 0 and 1+1e-8
+
+        Returns
+        -------
+        xarray.Dataset or xarray.DataArray
+            Histogram of forecast probabilities when the event was observed
+        xarray.Dataset or xarray.DataArray
+            Histogram of forecast probabilities when the event was not observed
+
+        Examples
+        --------
+        >>> observations = xr.DataArray(np.random.normal(size=(30,30)),
+        ...                             coords=[('x', np.arange(30)),
+        ...                                     ('y', np.arange(30))])
+        >>> forecasts = xr.DataArray(np.random.normal(size=(30,30,10)),
+        ...                          coords=[('x', np.arange(30)),
+        ...                                  ('y', np.arange(30)),
+        ...                                  ('member', np.arange(10))])
+        >>> forecast_event_likelihood = (forecasts > 0).mean('member')
+        >>> observed_event = observations > 0
+        >>> hist_event, hist_no_event = discrimination(observed_event, forecast_event_likelihood, dim=['x','y'])
+
+        Notes
+        -----
+        See http://www.cawcr.gov.au/projects/verification/
+    """
+
+    if dim is not None:
+        if len(dim) == 0:
+            raise ValueError(
+                'At least one dimension must be supplied to compute discrimination over'
+            )
+
+    hist_event = histogram(
+        forecasts.where(observations),
+        bins=[probability_bin_edges],
+        bin_names=[FORECAST_PROBABILITY_DIM],
+        bin_dim_suffix='',
+        dim=dim,
+    ) / (observations).sum(dim=dim)
+
+    hist_no_event = histogram(
+        forecasts.where(xr.ufuncs.logical_not(observations)),
+        bins=[probability_bin_edges],
+        bin_names=[FORECAST_PROBABILITY_DIM],
+        bin_dim_suffix='',
+        dim=dim,
+    ) / (xr.ufuncs.logical_not(observations)).sum(dim=dim)
+
+    return hist_event, hist_no_event
```

Comment:
@dougiesquire discrimination is the only function returning two objects. In doppyo you used a dataset here. If we wanted to change we could create a new dimension event = yes or no.